### DISCUSSION TITLE
Handle multi states uptime card

### DISCUSSION THREAD

**Discussion by dylandoamaral** [dylandoamaral/uptime-card#67] [Improvements]
Today, the card is too limited with the only two state possibilities. A sensor can be ON or OFF and only binary sensors works. However we can imagine a sensor with 3 states or more and the card should handle them. It should also handle non binary sensor if the user provide a function to transform the current value into a state such as: `sensor > 10 then state Loading`.

We should have the possibilities to choose the color for each state.

The only problem is state "in between", such as #53 where the user want to define threshold between the states, but what happen if there is three state, maybe setup a strategy option such as (the stronger state win, we blend the color according to the value) and so on.

Could somebody tell me if it can be useful for him or not ?

**Comment by dylandoamaral** [comment]
@agneevX what do you think about it, and how can I possibly make this feature work with #53 ?

**Comment by agneevX** [comment]
Thanks for the A2A. 

> sensor > 10 then state Loading

I would say this would be kind of complicated. Generally speaking, your card revolves around binary states ("uptime"), you've already made that possible.

But to map individual values of states, with operators, into a graph that is by default, binary would be, in my opinion, beyond the scope of your project!

So, to map more than two states, I'd suggest `sensor = 10 then state Loading` instead of `sensor > 10 then state Loading`.

E.g. 

```yaml
state_map:
  - state: 'on'
    value: Online
  - state: 'off'
    value: Offline
  - state: fix_in_progress
    value: Working
```

This would may override `ok`/`ko` options and alias states if set.

Again, I think going further than binary states should not be necessary, so what I asked was the ability to customize the display of binary states and set custom levels.

**Comment by dylandoamaral** [comment]
I think you are right, I personally only use this card with binary sensors and I guess trying to do everything with one card is not the good philosophy. I will still let that discussion open if other people want to argue in favor of the feature. But for the moment I am considering this one as `aborted`.

**Comment by agneevX** [comment]
I know this is off topic, but would you be able to create a PR for haptic support for `mini-graph-card` and `rgb-light-card`, if possible? Figured that you have good experience in JS. Thanks!

**Comment by dylandoamaral** [comment]
> I know this is off topic, but would you be able to create a PR for haptic support for `mini-graph-card` and `rgb-light-card`, if possible? Figured that you have good experience in JS. Thanks!

If I find the time and inclination to do so I will see to make a PR for `rgb-light-card` but it will be quit annoying since they don't extend  https://github.com/custom-cards/custom-card-helpers which provide built-in support for that. For `mini graph card`, there are many PRs and nothing seems to move anymore, the author asked the community to maintain the project but from then nobody seems to care/validate PRs.

**Comment by LRvdLinden** [comment]
@dylandoamaral Hi, is it possible to put a glance to uptime card? A red blink when it is down

**Comment by dylandoamaral** [comment]
Hi rise another issue please. Do you want the state to blink ? Or à round juste before the state that blink ? (Answer in the new issue please :D)

**Comment by LRvdLinden** [comment]
@dylandoamaral The blink you add is perfect. Thanks for all the hard work!
